core/src/main/java/com/nextdoor/bender/handler/HandlerMetadata.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright $year Nextdoor.com, Inc
+ */
+
+package com.nextdoor.bender.handler;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+/**
+ * The {@link HandlerMetadata} object is a simple Key/Value map that is used by {@Link
+ * BaseHandler} to store information about how and when the function was invoked. This data is
+ * either thrown away at the end of the execution, or is used later by our
+ * {@link com.nextdoor.bender.wrapper.Wrapper} and
+ * {@link com.nextdoor.bender.operation.Operation} classes to futher modify the payload before
+ * its shipped.
+ */
+public class HandlerMetadata {
```

Comment:
Add an `equals` and `hashCode` methods. These can just wrap `this.payload.equals` (though add a check for the immutable flag) and `this.payload.hashCode`.